### PR TITLE
optimize autotest document based on #891

### PR DESCRIPTION
**Pull request by HuangJiameng**
- fix contents according to suggestions in https://github.com/deepmodeling/dpgen/pull/891
- reorganize the document structure
- this doc is based on autotest wiki and README(devel)

### CODE REVIEW

**Comment by amcadmus** [comment]
@HuangJiameng do you want to replace #891 by this PR?  If so I will close #891

**Comment by HuangJiameng** [comment]
> @HuangJiameng do you want to replace #891 by this PR? If so I will close #891

Yes. Thanks.

File: doc/autotest/property/properties/Surface-make.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Surface make
+
+**Step 1.** Based on the equilibrium configuration, `generate_all_slabs` module in [pymatgen.core.surface](https://pymatgen.org/pymatgen.core.surface.html) would help to generate surface structure list with using `max_miller`, `min_slab_size`, and `min_vacuum_size` parameters. 
+
+**Step 2.** If `refine` is True, we do [refine process](https://github.com/deepmodeling/dpgen/wiki/Refine:-get-started-and-input-examples). If `reprod-opt` is True (the default is False), we do [reproduce process](https://github.com/deepmodeling/dpgen/wiki/Reproduce:-get-started-and-input-examples). Otherwise, the surface structure (`POSCAR`) with perturbations in xz and miller index information (`miller.out`) are written in the task directory, for example, in `confs/mp-*/interstitial_00/task.000000` with the check and possible removing of the old input files like before.
```
Comment (njzjz) [inline doc/autotest/property/properties/Surface-make.md:5]: Please update the links.

File: doc/autotest/property/properties/Surface-make.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+**Step 3.** In `surface` by VASP, if `change_box` is True (default value is True), `ISIF=4`, if False, `ISIF=2`, and if `static-opt` is True, `NSW=0`. For `surface` by LAMMPS, when `static-opt` is False, `in.lammps` is the same as that in [Relaxation: make](https://github.com/deepmodeling/dpgen/wiki/Relaxation:-make) when `change_box` is True or the same as that in [EOS: make](https://github.com/deepmodeling/dpgen/wiki/EOS:-make) when `change_box` is False. When `static-opt` is True, the same `in.lammps` as that in `VASP to LAMMPS` of [Reproduce: make](https://github.com/deepmodeling/dpgen/wiki/Reproduce:-make) would be written.
```
Comment (njzjz) [inline doc/autotest/property/properties/Surface-make.md:7]: Please update the links.

File: doc/autotest/property/properties/Interstitial-make.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## Interstitial make
+
+**Step 1.** For each element in `insert_ele` list, `InterstitialGenerator` module in [pymatgen.analysis.defects.generators](https://pymatgen.org/pymatgen.analysis.defects.generators.html) would help to generate interstitial structure. The structure would be appended into a list if it can meet the requirements in `conf_filters`.
+
+**Step 2.** If `refine` is `True`, we do [refine process](https://github.com/deepmodeling/dpgen/wiki/Refine:-get-started-and-input-examples). If `reprod-opt` is `True` (the default is **False**), we do [reproduce process](https://github.com/deepmodeling/dpgen/wiki/Reproduce:-get-started-and-input-examples). Else, the vacancy structure (`POSCAR`) and supercell information (`supercell.out`) are written in the task directory, for example, in `confs/mp-*/interstitial_00/task.000000` with the check and possible removing of the old input files like before.
+
+**Step 3.** In `interstitial` by VASP, `ISIF = 3`. In `interstitial` by LAMMPS, the same `in.lammps` as that in [EOS (change_box is True)](https://github.com/deepmodeling/dpgen/wiki/EOS:-make) would be generated with `scale` set to one.
```
Comment (njzjz) [inline doc/autotest/property/properties/Interstitial-make.md:7]: Please update the links.

File: doc/autotest/property/properties/Interstitial-make.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+## Interstitial make
+
+**Step 1.** For each element in `insert_ele` list, `InterstitialGenerator` module in [pymatgen.analysis.defects.generators](https://pymatgen.org/pymatgen.analysis.defects.generators.html) would help to generate interstitial structure. The structure would be appended into a list if it can meet the requirements in `conf_filters`.
+
+**Step 2.** If `refine` is `True`, we do [refine process](https://github.com/deepmodeling/dpgen/wiki/Refine:-get-started-and-input-examples). If `reprod-opt` is `True` (the default is **False**), we do [reproduce process](https://github.com/deepmodeling/dpgen/wiki/Reproduce:-get-started-and-input-examples). Else, the vacancy structure (`POSCAR`) and supercell information (`supercell.out`) are written in the task directory, for example, in `confs/mp-*/interstitial_00/task.000000` with the check and possible removing of the old input files like before.
```
Comment (njzjz) [inline doc/autotest/property/properties/Interstitial-make.md:5]: Please update the links.

File: doc/autotest/property/properties/Elastic-make.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+## Elastic make
+
+**Step 1.** The `DeformedStructureSet` module in [pymatgen.analysis.elasticity.strain](https://pymatgen.org/pymatgen.analysis.elasticity.strain.html) is used to generate a set of independently deformed structures. `equi.stress.out` file is written to record the equilibrium stress in the Elastic directory. For the example in the previous section, `equi.stress.out` should be in `confs/mp-*/elastic_00`.
+
+**Step 2.** If there are `init_from_suffix` and `output_suffix` parameter in the `properties` part, the [refine process](https://github.com/deepmodeling/dpgen/wiki/Refine:-get-started-and-input-examples) follows. Else, the deformed structure (`POSCAR`) and strain information (`strain.out`) are written in the task directory, for example, in `confs/mp-*/elastic_00/task.000000`.
```
Comment (njzjz) [inline doc/autotest/property/properties/Elastic-make.md:5]: Please update the links.

File: doc/autotest/property/properties/EOS-get-started-and-input-examples.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+## EOS get started and input examples
+
+Equation of State (EOS) here calculates the energies of the most stable structures as a function of volume. Users can refer to Figure 4 of the [dpgen JCPC paper](https://www.sciencedirect.com/science/article/pii/S001046552030045X?via%3Dihub) for more information of EOS.
```
Comment (njzjz) [inline doc/autotest/property/properties/EOS-get-started-and-input-examples.md:3]: CPC

File: doc/autotest/property/properties/Vacancy-make.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+## Vacancy make
+
+**Step 1.** The `VacancyGenerator` module in [pymatgen.analysis.defects.generators](https://pymatgen.org/pymatgen.analysis.defects.generators.html) is used to generate a set of structures with vacancy. 
+
+**Step 2.** If there are `init_from_suffix` and `output_suffix` parameter in the `properties` part, the [refine process](https://github.com/deepmodeling/dpgen/wiki/Refine:-get-started-and-input-examples) follows. If reproduce is evoked, the [reproduce process](https://github.com/deepmodeling/dpgen/wiki/Reproduce:-get-started-and-input-examples) follows. Otherwise, the vacancy structure (`POSCAR`) and supercell information (`supercell.out`) are written in the task directory, for example, in `confs/mp-*/vacancy_00/task.000000` with the check and possible removing of the old input files like before.
```
Comment (njzjz) [inline doc/autotest/property/properties/Vacancy-make.md:5]: Please update the links.

File: doc/autotest/property/properties/Vacancy-make.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+**Step 3.** When doing `vacancy` by VASP, `ISIF = 3`. When doing `vacancy` by LAMMPS, the same `in.lammps` as that in [EOS (change_box is True)](https://github.com/deepmodeling/dpgen/wiki/EOS:-make) would be generated with `scale` set to one.
```
Comment (njzjz) [inline doc/autotest/property/properties/Vacancy-make.md:7]: Please update the links.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: doc/autotest/property/properties/Elastic-make.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
+## Elastic make
+
+**Step 1.** The `DeformedStructureSet` module in [pymatgen.analysis.elasticity.strain](https://pymatgen.org/pymatgen.analysis.elasticity.strain.html) is used to generate a set of independently deformed structures. `equi.stress.out` file is written to record the equilibrium stress in the Elastic directory. For the example in the previous section, `equi.stress.out` should be in `confs/mp-*/elastic_00`.
+
+**Step 2.** If there are `init_from_suffix` and `output_suffix` parameter in the `properties` part, the [refine process](../../refine/Refine-get-started-and-input-examples.html) follows. Else, the deformed structure (`POSCAR`) and strain information (`strain.out`) are written in the task directory, for example, in `confs/mp-*/elastic_00/task.000000`.
```
Comment (njzjz) [inline doc/autotest/property/properties/Elastic-make.md:5]: This link does not work.

image

Comment (njzjz) [reply doc/autotest/property/properties/Elastic-make.md:5]: Please also check other links.

**Review by njzjz — CHANGES_REQUESTED** [review]

**Comment by njzjz** [comment]
It looks like the links are still unavailable.

**Comment by HuangJiameng** [comment]
> It looks like the links are still unavailable.

Hmm... They work well when I check them.

**Comment by njzjz** [comment]
> > It looks like the links are still unavailable.
> 
> Hmm... They work well when I check them.

Oh, I was wrong.

**Review by njzjz — APPROVED** [review]
LGTM

File: doc/autotest/property/properties/EOS-get-started-and-input-examples.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+			"vol_start":	10,
+			"vol_end":	30,
+			"vol_step":	0.5,
+                        "change_box":   true
+		}
+        ]
+}
+```
+
+`vol_start` is the starting volume per atom in Å^3/atom, `vol_step` is the increasing step of volume and the biggest volume is smaller than `vol_end`. In the above example, 40 tasks would be generated as `task.000000` to `task.000039` with the volume `10.00, 10.50, 11.00, ..., 29.50` Å^3/atom, respectively.
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-get-started-and-input-examples.md:28]: This EOS example is not for the current autotest. The content needs to be consistent with the example in Propert-get-started-and-examples.md.

File: doc/autotest/property/properties/EOS-make.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+
+```txt
+scale = (vol_current / vol_equi) ** (1. / 3.)
+```
+
+`vol_current` is the corresponding volume per atom of the current task and `vol_equi` is the volume per atom of the equilibrium configuration. Then the `poscar_scale` function in `dpgen.auto_test.lib.vasp` module would help to generate `POSCAR` file with `vol_current` in `confs/mp-*/eos_00/task.[0-9]*[0-9]`.
+
+**Step 5.** According to the task type, the input file including `INCAR`, `POTCAR` or `conf.lmp`, `in.lammps` would be written in every `confs/mp-*/eos_00/task.[0-9]*[0-9]`. 
+
+For EOS calculations by VASP, if `change_box` is `True`, `ISIF` in VASP would be 4, else `ISIF` would be 2. The default value of `change_box` is `True`. For further information of the use of `ISIF` in VASP, we refer users to [ISIF command](https://www.vasp.at/wiki/index.php/ISIF).
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-make.md:19]: We do not have the keyword "change_box" now.

File: doc/autotest/property/properties/EOS-post.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+```txt
+conf_dir: confs/mp-100/eos_00
+ VpA(A^3)  EpA(eV)
+ 10.000   -3.0245
+ 10.500   -3.0216
+  ...        ...
+ 29.500   -7.9740
+```
+
+
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-post.md:35]: These are the outputs for the old autotest.

File: doc/autotest/property/properties/Elastic-get-started-and-input-examples.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+            {
+                "type":         "elastic",
+                "norm_deform":	2e-2,
+	       "shear_deform":  5e-2
+	    }
+        ]
+}
+```
+
+Here the default values of `norm_deform` and `shear_deform` are **2e-3** and **5e-3**, respectively. A list of `norm_strains` and `shear_strains` would be generated as below:
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Elastic-get-started-and-input-examples.md:25]: We have changed the default value for deformation. Please look at the example at Property-get-started-and-examples.md.

File: doc/autotest/property/properties/Elastic-make.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+mass            2 1
+neigh_modify    every 1 delay 0 check no
+pair_style deepmd frozen_model.pb
+pair_coeff
+compute         mype all pe
+thermo          100
+thermo_style    custom step pe pxx pyy pzz pxy pxz pyz lx ly lz vol c_mype
+dump            1 all custom 100 dump.relax id type xs ys zs fx fy fz
+min_style       cg
+minimize        1.000000e-12 1.000000e-06 5000 500000
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Elastic-make.md:26]: We have changed the default value for lammps to 0 and 1e-10.

File: doc/autotest/property/properties/Interstitial-post.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+}
+```
+
+#### result.out
+```txt
+conf_dir: confs/mp-100/interstitial_00
+Insert_ele-Struct: Inter_E(eV)  E(eV) equi_E(eV)
+task.000000:   1.137  -297.162 -298.299
+task.000001:   1.600  -296.699 -298.299
+```
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Interstitial-post.md:21]: This output is not consistent with Property-post.md.

File: doc/autotest/property/properties/Surface-post.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+}
+```
+
+#### result.out
+```txt
+conf_dir: confs/mp-100/surface_00
+Miller_Indices:     Surf_E(J/m^2) EpA(eV) equi_EpA(eV)
+[1,1,1]:       1.200  -297.099 -298.299
+[1,2,1]:       1.400  -296.899 -298.299
+```
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Surface-post.md:22]: This output is not consistent with Property-post.md.

File: doc/autotest/property/properties/Vacancy-post.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+}
+```
+
+#### result.out
+```txt
+conf_dir: confs/mp-100/vacancy_00
+Structure: 	Vac_E(eV)  E(eV) equi_E(eV)
+task.000000:   2.137  -296.162 -298.299
+task.000001:   1.500  -296.799 -298.299
+```
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Vacancy-post.md:21]: This output is not consistent with Property-post.md.

File: doc/autotest/property/properties/EOS-make.md [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+pair_coeff
+compute         mype all pe
+thermo          100
+thermo_style    custom step pe pxx pyy pzz pxy pxz pyz lx ly lz vol c_mype
+dump            1 all custom 100 dump.relax id type xs ys zs fx fy fz
+min_style       cg
+fix             1 all box/relax iso ${Px}
+minimize        1.000000e-12 1.000000e-06 5000 500000
+fix             1 all box/relax aniso ${Px}
+minimize        1.000000e-12 1.000000e-06 5000 500000
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-make.md:51]: We have changed the default value for lammps to 0 and 1e-10.

File: doc/autotest/property/properties/EOS-make.md [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+mass            2 1
+neigh_modify    every 1 delay 0 check no
+pair_style deepmd frozen_model.pb
+pair_coeff
+compute         mype all pe
+thermo          100
+thermo_style    custom step pe pxx pyy pzz pxy pxz pyz lx ly lz vol c_mype
+dump            1 all custom 100 dump.relax id type xs ys zs fx fy fz
+min_style       cg
+minimize        1.000000e-12 1.000000e-06 5000 500000
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-make.md:90]: We have changed the default value for lammps to 0 and 1e-10.

File: doc/autotest/property/properties/Elastic-post.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+ 112.91  132.64   97.92    0.00    0.00   -0.00
+  96.10   96.10  263.48    0.00    0.00    0.00
+   0.00   -0.00   -0.00   33.50    0.00    0.00
+   0.00   -0.00   -0.00    0.00   33.29    0.00
+   0.00    0.00   -0.00   -0.00   -0.00   18.70
+# Bulk   Modulus BV = 126.75 GPa
+# Shear  Modulus GV = 31.57 GPa
+# Youngs Modulus EV = 87.46 GPa
+# Poission Ratio uV = 0.38
+```
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Elastic-post.md:36]: This output is not consistent with Property-post.md.

File: doc/autotest/property/properties/Surface-get-started-and-input-examples.md [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+                "type":           "surface",
+                "min_slab_size":  10,
+                "min_vacuum_size":11,
+                "pert_xz":        0.01,
+                "max_miller":     1,
+                "static-opt":     false 
+	    }
+        ]
+}
+```
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/Surface-get-started-and-input-examples.md:26]: This example is not consistent with Property-get-started-and-examples.md.

File: doc/autotest/property/properties/EOS-make.md (hunk 11 above)
Comment (HuangJiameng) [reply doc/autotest/property/properties/EOS-make.md:19]: https://github.com/deepmodeling/dpgen/blob/0767dce7cad29367edb2e4a55fd0d8724dbda642/dpgen/auto_test/lib/lammps.py#L217

But it seems still work in the code.

Comment (kevinwenminion) [reply doc/autotest/property/properties/EOS-make.md:19]: Yes, "change_box" still exists in lib/lammps.py. However, we do not have "change_box" parameter in EOS.py. In EOS make, if we do not explicitly set "vol_abs" to be true, the "vol_start" and "vol_end" is for the relative volume with respect to the equilibrium volume. So the current example will lead to the volume start = volume equilibrium * 10 and this is not we want.

File: doc/autotest/property/properties/EOS-get-started-and-input-examples.md [hunk 22]
```diff
@@ -14,15 +14,14 @@ Equation of State (EOS) here calculates the energies of the most stable structur
...
+         "vol_start":    0.9,
+         "vol_end":      1.1,
+         "vol_step":     0.01
+        }
         ]
 }
 ```
 
-`vol_start` is the starting volume per atom in Å^3/atom, `vol_step` is the increasing step of volume and the biggest volume is smaller than `vol_end`. In the above example, 40 tasks would be generated as `task.000000` to `task.000039` with the volume `10.00, 10.50, 11.00, ..., 29.50` Å^3/atom, respectively.
+`vol_start` is the starting volume per atom in Å^3/atom, `vol_step` is the increasing step of volume, and the biggest volume is smaller than `vol_end`. In the above example, 40 tasks would be generated as `task.000000` to `task.000020` with from 0.9 to 1.1 times the volume of equilibrium structure (Å^3/atom as the unit), respectively.
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-get-started-and-input-examples.md:27]: `vol_start` and `vol_end` are relative volume by default.

File: examples/test/vasp_param_from_incar.json [hunk 23]
```diff
@@ -37,7 +37,6 @@
     "_comment": "pert xz to work around vasp bug...",
     "pert_xz": 0.01,
     "max_miller": 2,
-    "static-opt": false,
     "relax_box": false,
     "_comment": "06.phonon",
     "supercell_matrix": [
```
Comment (kevinwenminion) [inline examples/test/vasp_param_from_incar.json:42]: These examples are for the old autotest. The input examples for the current autotest are already in doc. We can considering replacing these old examples (maybe in another pr).

File: doc/autotest/property/properties/EOS-get-started-and-input-examples.md (hunk 22 above)
Comment (HuangJiameng) [reply doc/autotest/property/properties/EOS-get-started-and-input-examples.md:27]: Yes, so what's wrong?

File: examples/test/vasp_param_from_incar.json (hunk 23 above)
Comment (HuangJiameng) [reply examples/test/vasp_param_from_incar.json:42]: OK. Looking forward to it.

File: doc/autotest/property/properties/EOS-get-started-and-input-examples.md [hunk 24]
```diff
@@ -24,4 +24,4 @@ Equation of State (EOS) here calculates the energies of the most stable structur
 }
 ```
 
-`vol_start` is the starting volume per atom in Å^3/atom, `vol_step` is the increasing step of volume, and the biggest volume is smaller than `vol_end`. In the above example, 40 tasks would be generated as `task.000000` to `task.000020` with from 0.9 to 1.1 times the volume of equilibrium structure (Å^3/atom as the unit), respectively.
+`vol_start` is the starting relative volume per atom in Å^3/atom, `vol_step` is the increasing step of the ratio of relative volume, and the biggest volume is smaller than `vol_end` times the volume of equilibrium structure.
```
Comment (kevinwenminion) [inline doc/autotest/property/properties/EOS-get-started-and-input-examples.md:27]: How about this:

`vol_start` is the starting volume relative to the equilibrium structure, `vol_step` is the volume increment step relative to the equilibrium structure, and the biggest relative volume is smaller than `vol_end`.

Comment (HuangJiameng) [reply doc/autotest/property/properties/EOS-get-started-and-input-examples.md:27]: LGTM

**Pull request merged**